src/helper/utils.ts

```diff
@@ -14,11 +14,22 @@
  * limitations under the License.
  */
 
+import { SimpleArray } from 'immutable-class';
+
 var objectHasOwnProperty = Object.prototype.hasOwnProperty;
 export function hasOwnProperty(obj: any, key: string): boolean {
   return objectHasOwnProperty.call(obj, key);
 }
 
+export function findProperty(obj: any, key: string): any {
+  return hasOwnProperty(obj, key) ? key : null;
+}
+
+export function findPropertyCI(obj: any, key: string): any {
+  var lowerKey = key.toLowerCase();
+  return SimpleArray.find(Object.keys(obj), (v) => v.toLowerCase() === lowerKey);
+}
```

Comment:
considering where they are used I think findProperty and findPropertyCI could be static on RefExpression